### PR TITLE
feat: context menu handler.

### PR DESCRIPTION
**Pull request by itsUrcute**
> **Context menu handler**
- handle context menu items in `src/context-commands`.
- change types of slash command objects and event objects for more precise and consistent code.

> **Extra features**
- `start:no-edit`: NPM script to run the bot without setting the commands, hence reducing the number of API calls when no command edits are performed.
- `start:type-change`: NPM script to run the bot when you change the type of a command.

### CODE REVIEW

**Review by TogarashiPepper — APPROVED** [review]

File: src/internals/commandHandler.ts [hunk 1]
```diff
@@ -49,47 +64,58 @@ function startCommandHandler(client: Client): Collection<string, Command> {
...
+			// Setting the commands as slash commands in the selected guild.
+			commandData = await setCommands(slashCommands, contextCommands);
+
+			// Configuring permissions for every command.
+			await setPermissions(slashCommands, commandData);
+		} else {
+			console.log("No commands set.");
+		}
 	});
 	client.on("interactionCreate", (interaction) => {
```
Comment (cheeze2000) [inline src/internals/commandHandler.ts:102]: any reason why this isn't in `events/general/interactionCreate` instead?

File: src/internals/commandHandler.ts [hunk 2]
```diff
@@ -49,47 +64,58 @@ function startCommandHandler(client: Client): Collection<string, Command> {
...
-		if (!interaction.isCommand()) {
-			return;
-		}
-		const cmd = commands.get(interaction.commandName);
-		if (!cmd) {
-			return;
+		if (interaction.isCommand()) {
```
Comment (cheeze2000) [inline src/internals/commandHandler.ts:103]: is it possibel to use something like `slashCommands.get(interaction.commandname)?.execute(interaction)`?
same goes for line 110

File: src/internals/commandHandler.ts (hunk 1 above)
Comment (itsUrcute) [reply src/internals/commandHandler.ts:102]: It's in here for the purpose of isolating internal handling and user code.

File: src/internals/commandHandler.ts (hunk 2 above)
Comment (itsUrcute) [reply src/internals/commandHandler.ts:103]: This format is more suitable for future internal checks.

**Pull request merged**